packages/core/core/src/public/MainAssetGraph.js

```diff
@@ -38,6 +38,12 @@ export default class MainAssetGraph implements IMainAssetGraph {
       to: assetNode.id
     });
 
+    graph.traverseAssets(currentAsset => {
```

Comment:
We now prune assets that don't match the bundle type when cloning the asset's subtree in the main asset graph.

This could be done in the creation of the subgraph itself, but given @padmaia's work on new node types which might include the creation of a leaner graph, I didn't want to create a bunch of unnecessary work. This can and should probably change down the road.